src/components/Activities/ConfirmSubmitPopover.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useRef } from 'react';
+import { Button, Overlay, Popover, Spinner } from 'react-bootstrap';
+import { useOutsideClickDetect } from '../../hooks/useOutsideClickDetect';
+
+export interface IConfirmSubmitPopoverProps {
+    show: boolean,
+    target: any,
+    title: string,
+    text: string,
+    submitting: boolean,
+    buttonVariant: "primary" | "danger",
+    handleClosePopoverClick: () => void,
+    handleSubmit: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void
+}
+
+export const ConfirmSubmitPopover: React.FunctionComponent<IConfirmSubmitPopoverProps> = (props) => {
+
+    const wrapperRef = useRef(null);
+    useOutsideClickDetect(wrapperRef, props.handleClosePopoverClick);
+
+    return (
+        <Overlay
+            show={props.show}
+            placement="top"
+            target={props.target}
+        >
+            <Popover id={"delete-popover"}>
+                <div ref={wrapperRef}>
+                    <Popover.Title as="h3">{props.title}</Popover.Title>
+                    <Popover.Content>
+                        <p>{props.text}</p>
+                        <Button
+                            className="float-left mb-2"
+                            variant="secondary"
+                            onClick={() => props.handleClosePopoverClick()}
+                        >
+                            Close
+                        </Button>
+                        <Button
+                            className="float-right mb-2"
+                            disabled={props.submitting}
+                            variant={props.buttonVariant}
+                            onClick={(e) => props.handleSubmit(e)}
```

Comment:
I get an error for an implicit 'any' on this line as well as a  few other onclick/onchange/onsubmit events without any typing. Can you look for all the (e) and give them an explicit type? I'm semi-okay with any, but if we could get the proper TS types that'd be awesome!
I think for update e: React.ChangeEvent for submit: e: React.FormEvent for click: e: React.MouseEvent (though I could be wrong on any/all of these)

Reply:
I'm an idiot. forgot to npm install to get the type definitions